api/graphql/schema.graphql

```diff
@@ -594,7 +594,7 @@ input CreateUserInput {
     # Personal info
     firstName: String! @rename(attribute: "first_name")
     lastName: String! @rename(attribute: "last_name")
-    email: Email! @rules(apply: ["unique:users,email"])
```

Comment:
+1 to keeping the uniqueness of emails. Even though we don't technically need it, feels really against the grain to allow multiple emails for accounts, without us kinda "clearing" that as sensible.

The value of releasing a constrain doesn't feel worth losing the sanity-check :) Seems like it could get messy

Reply:
I removed it because it seemed like this special auth-specific handling.  We don't have the constraint on `telephone`, for example.  

We actually still have the database constraint in place for email so it is enforcing uniqueness there.  Is this directive redundant?  Should it be moved to sub?  Should we have the directive on both fields so they match the database?  Should we remove the database constraint on email altogether?  Will we have edge cases where multiple users want to use a shared mailbox or distro list?  If they end up with multiple accounts that need to be merged matching on email will be a nice trick.

I agree that this is inconsistent as it is but I'm not sure where we want to fall on consistency.

Reply:
1. I believe our schema rules should match the database. I think it makes the constraints (and the errors if they're violated) more visible.
2. I'm leaning towards keeping the email uniqueness constraint. Eventually we may probably want to confirm emails, and if emails are unique that would help avoid spam accounts. Merging accounts should be a rare and exceptional event, and Admins could modify the old account if needing to free up the email again.

Reply:
Yeah, I like that.  I've added the schema validation back in for email and added one for sub as well.

Reply:
> I'm leaning towards keeping the email uniqueness constraint. Eventually we may probably want to confirm emails

Similar vibes. It feels possible that once running, we might learn about some pragmatic downsides to treating email as unverified contact details (it seems like a very ubiquitous pattern, even when proliferation of oauth providers mean it's not required). Cautiousness would seem to avoid heavier corrective interactions with actual users, if that's needed (which is the sort of thing that might paralyze a decision 🙂 )